### DISCUSSION TITLE
404 error on deployment

### DISCUSSION THREAD

**Discussion by ApexSQ** [vercel/community#7444] [Help] [answered]
### Summary

i get this error in logs 

GET

/Index.js
connected to database
Legacy server listening.

backend code Index.js

const express = require('express');
const cors = require("cors");
const mongoose = require('mongoose');
const app = express();

// setting up middleware
app.use(express.json())
app.use(express.urlencoded({ extended : true }))
app.use(cors(*));

// importing model
const ListAppModel = require("./model/ListApp");

const ConnectionString = "mongodb+srv://djosh734:.mongodb.net/ListAppDB";

mongoose.connect(ConnectionString).then(() => {
  console.log("connected to database");

  const port = process.env.PORT || 3000;
  app.listen(port, function() {
    console.log(`Server running on port ${port}`);
  });

})
.catch((error) => {
  console.log('Error connecting to database:', error);
});

// CRUD Operations

app.get('/', (req, res) => {
  res.send('Welcome to the API!');
});

// Get Method
app.get("/lists", async (req, res) => {
  try {
    const response = await ListAppModel.find({});
    console.log(response);
    res.json(response);
  } catch (error) {
    console.log(error);
    res.status(500).json({ message: 'Error fetching data', error: error.message });
  }
});

// Create Method PostMethod

app.post("/lists", async (req, res) => {
  try {

    console.log(req.body);

    const list = req.body;
   
    const newItem = await ListAppModel.create(list);

    res.status(200).send( "successfull");
    
  } catch (error) {
    console.log(error)
    res.status(500).json({ message: 'Error fetching data', error: error.message });
  }

});

//delete Method

app.delete("/lists/:id", async (req, res) => {
  try {

    let id = req.params.id;
    console.log(id);

    const deleteItem = await ListAppModel.deleteOne({
      _id: id
    });

    res.status(200).send( " DELETE success")

    
    
  } catch (error) {
    console.log(error)
    res.status(500).json({ message: 'Error fetching data', error: error.message });
  }

});

// put Method

app.put("/lists/:id", async (req, res) => {
  try {
    const id = req.params.id;
    const { text, status } = req.body;

    const updatedItem = await ListAppModel.findByIdAndUpdate(
      id,
      { $set: { text: text, status: status } },
      { new: true, runValidators: true }
    );

    if (!updatedItem) {
      return res.status(404).json({ message: 'Item not found' });
    }

    res.status(200).json(updatedItem);
  } catch (error) {
    console.error(error);
    res.status(500).json({ message: 'Error updating item', error: error.message });
  }
});

module.exports = app;

/////

vercel.json file

{
  "version": 2,
  "builds": [
    {
      "src": "Index.js",
      "use": "@vercel/node"
    }
  ],
  "routes": [
    {
      "src": "/(.*)",
      "dest": "/"
    }
  ],
  "env": {
    "NODE_DEBUG": "true"
  }
}

### Steps to Reproduce

on deployment

**Comment by pawlean** [comment]
Hi, @ApexSQ!

Here are the main issues I see from the output above:

1. The MongoDB connection string is incomplete and does not include the password.
2. The server running message uses incorrect string concatenation instead of template literals.
3. The `src` path in `vercel.json` might not correctly point to `Index.js`, and the `dest` path might need to be more specific.

Let us know how you get on!

**Reply by ApexSQ** [reply]
thank you for responding 

1.MongoDB connection string is like that for security it is correct on my end I can connect and use the server locally.

would I use the whole path for index.js I have tried a variety of paths all have the same issue.

please let me know what you think

**Comment by ApexSQ** [comment, marked as answer]
Because my backend was Named Index.js with a capital I i need to specify the path as shown below 

{
  
  "version": 2,
  "builds": [{ "src": "*.js", "use": "@vercel/node" }],
  "routes": [
      { "src": "/.*", "dest": "/Index.js" }
  ]

}

[a comment by an automated account is not shown]
